test/mitmproxy/tools/web/test_app.py

```diff
@@ -480,6 +480,26 @@ def test_xsrf_hardening_login(self):
             ).body
         )
 
+    def test_login_with_token_header(self):
+        web_password = self.master.addons.get("webauth")._password
+        headers = {"Cookie": "", "Authorization": f"Bearer {web_password}"}
+        assert self.fetch("/", headers=headers).code == 200
+
+    def test_login_with_token_param(self):
+        web_password = self.master.addons.get("webauth")._password
+        resp = self.fetch(f"/?token={web_password}", headers={"Cookie": ""})
+        assert resp.code == 200
+
+    def test_login_with_malformed_auth_header(self):
+        resp = self.fetch("/", headers={"Cookie": "", "Authorization": f"Bearer"})
+        assert resp.code == 403
+
+    def test_login_with_invalid_auth_header(self):
+        resp = self.fetch(
+            "/", headers={"Cookie": "", "Authorization": f"Bearer invalid_token"}
+        )
+        assert resp.code == 403
+
```

Comment:
Nit: Let's make sure the test style is consistent. Either always assign to `resp=`, or always do `assert self.fetch(...).code == 200`. Same for `headers`.

Reply:
Fixed with 7b04a12fda67d10b956cd886179aae3f377dd6fb